src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -206,6 +206,16 @@ def _stream_logs(byte_size,
     if curr_bytes:
         print(curr_bytes.decode('utf-8', errors='ignore'))
 
+    build_status = _get_build_status(metadata).lower()
+    logger.debug("Build status was: '{}'".format(build_status))
+
+    if build_status == 'internalerror' or build_status == 'failed':
```

Comment:
From the k-map is there any other condition - I might be missing something here. 
 build_status !== null && build_status.Status == "success" return 0 
Failure can return something like 
CLIError("Build status: '{}' with exit code ".format(build_status.status, build_status.exitCode)
I guess you might even be able to replace Complete with exitCode. 
Again I haven't through this through so better to have a way to remove complete with a single metadata field was the basic idea.

Reply:
@northtyphoon not sure if we want to error on show-logs, but that's the current behavior. To me it just adds more information; you can always discard the exit code if you don't plan to use it. Maybe we want a flag to disable errors though.

Reply:
I doubt if CLI has a consistent way to output meaningful exit code and in most cases raising CLIError is the suggested approach. Even if we can exit with a code, I would question how easy it is for users to understand the meaning of it. Given that we end up exiting with the same code, I think this path can be simplified to something like 
`raise CLIError("Build exited with status: {}".format(build_status))` if not successful.